### PR TITLE
Update illumiprocessor to v2.10

### PR DESCRIPTION
**Pull request by brantfaircloth**
Updating illumiprocessor code to v.2.10.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/illumiprocessor/meta.yaml [hunk 1]
```diff
@@ -7,20 +7,22 @@ package:
...
   url: https://github.com/faircloth-lab/{{ name }}/archive/v{{ version }}.tar.gz
-  sha256: 81a70360e43622d7ec73068d5d0fe79f7c82d7a8c50099b07e703431f220b1fd
+  sha256: 11694c25df15e0f75644a7d650476c2ddf3dbf83135b3ca5cd57a122804581a8
 
 build:
-  number: 2
+  number: 0
   noarch: python
   script: {{ PYTHON }} -m pip install . --ignore-installed --no-deps -vv
+  pin_depends: strict
```
Comment (bgruening) [inline recipes/illumiprocessor/meta.yaml:16]: Why is this needed here?

Comment (brantfaircloth) [reply recipes/illumiprocessor/meta.yaml:16]: It's not - I can remove.

Comment (brantfaircloth) [reply recipes/illumiprocessor/meta.yaml]: Fixed.

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge
